File: erpnext/accounts/report/team_wise_sales_report/team_wise_sales_report.js
frappe.query_reports["Team Wise Sales Report"] = {
    "filters": [
        {
            "fieldname": "from_date",
            "label": __("From Date"),
            "fieldtype": "Date",
            "reqd": 1
        },
        {
            "fieldname": "to_date",
            "label": __("To Date"),
            "fieldtype": "Date",
            "reqd": 1
        }
    ],

    onload: function(report) {
        // Show chart
        report.page.add_inner_button(__("Generate Graph"), function() {
            if (report.chart) {
                report.chart.parent.chart_options.valuesOverPoints = 1;
                report.chart.draw(true);
                report.chart.show();
            } else {
                frappe.msgprint(__("No chart data found"));
            }
        });

        // Print chart + table with values
        report.page.add_inner_button(__("Print with Graph"), function() {
            if (!report.chart) {
                frappe.msgprint(__("No chart data found to print"));
                return;
            }

            // Serialize SVG
            let serializer = new XMLSerializer();
            let svgNode = report.chart.svg;
            let svgString = serializer.serializeToString(svgNode);

            let canvas = document.createElement("canvas");
            let ctx = canvas.getContext("2d");
            let DOMURL = self.URL || self.webkitURL || window;

            let img = new Image();
            let svgBlob = new Blob([svgString], {type: "image/svg+xml;charset=utf-8"});
            let url = DOMURL.createObjectURL(svgBlob);

            img.onload = function() {
                canvas.width = img.width;
                canvas.height = img.height;
                ctx.drawImage(img, 0, 0);

                // Draw values over each point
                let datasets = report.chart.data.datasets;
                let labels = report.chart.data.labels;
                let chart_width = canvas.width;
                let chart_height = canvas.height;

                let padding = 40; // approx padding
                let maxVal = Math.max(...[].concat(...datasets.map(ds => ds.values)));

                datasets.forEach((ds, dsIndex) => {
                    ds.values.forEach((v, i) => {
                        let x = padding + (i / (labels.length - 1)) * (chart_width - 2*padding);
                        let y = chart_height - padding - (v / maxVal) * (chart_height - 2*padding);
                        ctx.fillStyle = dsIndex === 0 ? "#1E90FF" : "#28a745";
                        ctx.font = "bold 14px Arial";
                        ctx.textAlign = "center";
                        ctx.fillText(v.toFixed(0), x, y - 5);
                    });
                });

                let imgURL = canvas.toDataURL("image/png");
                DOMURL.revokeObjectURL(url);

                // Build table
                let tableHTML = '<table><thead><tr>';
                report.columns.forEach(col => {
                    tableHTML += `<th>${col.label}</th>`;
                });
                tableHTML += '</tr></thead><tbody>';

                report.data.forEach(row => {
                    tableHTML += '<tr>';
                    report.columns.forEach(col => {
                        let val = row[col.fieldname] || '';
                        tableHTML += `<td>${val}</td>`;
                    });
                    tableHTML += '</tr>';
                });
                tableHTML += '</tbody></table>';

                // Open print window
                let win = window.open("");
                win.document.write(`
                    <html>
                    <head>
                        <title>Team Wise Sales Report</title>
                        <style>
                            body { font-family: Arial, sans-serif; padding: 20px; }
                            table { border-collapse: collapse; width: 100%; margin-top: 20px; }
                            th, td { border: 1px solid #ccc; padding: 8px; text-align: left; }
                        </style>
                    </head>
                    <body>
                        <h2>Team Wise Sales Report</h2>
                        <img src="${imgURL}" style="max-width:100%; margin-bottom:20px;"/>
                        ${tableHTML}
                    </body>
                    </html>
                `);
                win.document.close();
                win.print();
            };
            img.src = url;
        });
    }
};

File: erpnext/accounts/report/team_wise_sales_report/team_wise_sales_report.py
import frappe
from frappe import _

def execute(filters=None):
    if not filters:
        filters = {}

    from_date = filters.get("from_date")
    to_date = filters.get("to_date")

    # Columns
    columns = [
        {"label": _("Month"), "fieldname": "month", "fieldtype": "Data", "width": 120},
        {"label": _("Blue Sales"), "fieldname": "blue_sales", "fieldtype": "Currency", "width": 150},
        {"label": _("Green Sales"), "fieldname": "green_sales", "fieldtype": "Currency", "width": 150},
        {"label": _("Total Sales"), "fieldname": "total_sales", "fieldtype": "Currency", "width": 150},
    ]

    # Data
    data = frappe.db.sql(
        """
        SELECT
            DATE_FORMAT(DATE(si.posting_date - INTERVAL DAY(si.posting_date)-1 DAY), '%%M %%y') AS month,
            SUM(CASE WHEN isplit.team = 'Blue'
                     THEN (isplit.team_amount / isplit.invoice_total_items) * si.base_grand_total
                     ELSE 0 END) AS blue_sales,
            SUM(CASE WHEN isplit.team = 'Green'
                     THEN (isplit.team_amount / isplit.invoice_total_items) * si.base_grand_total
                     ELSE 0 END) AS green_sales,
            SUM(si.base_grand_total) AS total_sales
        FROM `tabSales Invoice` si
        JOIN `tabCustomer` c ON si.customer = c.name
        JOIN (
            SELECT
                sii.parent AS sales_invoice,
                itm.belong_to AS team,
                SUM(sii.base_amount) AS team_amount,
                SUM(SUM(sii.base_amount)) OVER (PARTITION BY sii.parent) AS invoice_total_items
            FROM `tabSales Invoice Item` sii
            LEFT JOIN `tabItem` itm ON sii.item_code = itm.name
            GROUP BY sii.parent, itm.belong_to
        ) AS isplit ON si.name = isplit.sales_invoice
        WHERE si.docstatus = 1
          AND c.customer_group = 'Local'
          AND si.posting_date BETWEEN %(from_date)s AND %(to_date)s
        GROUP BY YEAR(si.posting_date), MONTH(si.posting_date)

        UNION ALL

        SELECT
            'Grand Total' AS month,
            SUM(CASE WHEN isplit.team = 'Blue'
                     THEN (isplit.team_amount / isplit.invoice_total_items) * si.base_grand_total
                     ELSE 0 END) AS blue_sales,
            SUM(CASE WHEN isplit.team = 'Green'
                     THEN (isplit.team_amount / isplit.invoice_total_items) * si.base_grand_total
                     ELSE 0 END) AS green_sales,
            SUM(si.base_grand_total) AS total_sales
        FROM `tabSales Invoice` si
        JOIN `tabCustomer` c ON si.customer = c.name
        JOIN (
            SELECT
                sii.parent AS sales_invoice,
                itm.belong_to AS team,
                SUM(sii.base_amount) AS team_amount,
                SUM(SUM(sii.base_amount)) OVER (PARTITION BY sii.parent) AS invoice_total_items
            FROM `tabSales Invoice Item` sii
            LEFT JOIN `tabItem` itm ON sii.item_code = itm.name
            GROUP BY sii.parent, itm.belong_to
        ) AS isplit ON si.name = isplit.sales_invoice
        WHERE si.docstatus = 1
          AND c.customer_group = 'Local'
          AND si.posting_date BETWEEN %(from_date)s AND %(to_date)s
        """,
        {"from_date": from_date, "to_date": to_date},
        as_dict=True,
    )

    # Chart data
    chart_data = {
        "data": {
            "labels": [d["month"] for d in data if d["month"] != "Grand Total"],
            "datasets": [
                {
                    "name": "Blue Sales",
                    "values": [float(d["blue_sales"] or 0) for d in data if d["month"] != "Grand Total"],
                },
                {
                    "name": "Green Sales",
                    "values": [float(d["green_sales"] or 0) for d in data if d["month"] != "Grand Total"],
                },
            ],
        },
        "type": "line",
        "colors": ["#1E90FF", "#28a745"],
        "valuesOverPoints": 1,  # show values on graph points
    }

    return columns, data, None, chart_data

